src/app/views/datasets/metadata/VersionMetadata.jsx

```diff
@@ -1,75 +1,352 @@
+          });
         }
 
-        this.setState({isVersion : true});
+        this.props.dispatch(updateActiveDataset(responses[2].current));
+
+        this.setState({
+          title: this.props.dataset.title,
+          release_frequency: this.props.dataset.release_frequency,
+          isFetchingData: false
+        });
+
+        }).catch(error => {
+            switch (error.status) {
+                case(403):{
+                    const notification = {
+                        "type": "neutral",
+                        "message": "You do not permission to view the edition metadata for this dataset",
+                        isDismissable: true
+                    }
+                    notifications.add(notification);
+                    break;
+                }
+                case (404): {
+                    const notification = {
+                        "type": "neutral",
+                        "message": `Dataset ID '${this.props.params.datasetID}' was not recognised. You've been redirected to the datasets home screen`,
+                        isDismissable: true
+                    };
+                    notifications.add(notification);
+                    this.props.dispatch(push(url.parent(url.parent())));
+                    break;
+                }
+                default: {
+                    const notification = {
+                        type: "warning",
+                        message: "An unexpected error's occurred whilst trying to get this dataset",
+                        isDismissable: true
+                    }
+                    notifications.add(notification);
+                    break;
+                }
+            }
+            console.error("Error has occurred:\n", error);
+        });
     }
 
     shouldComponentUpdate(_, nextState) {
-        // No need to re-render, this state does not impact the view. 
+        // No need to re-render, this state does not impact the view.
         // Just used to confirm whether we're on a version or instance
-        if (nextState.isVersion) {
+        if (this.props.isInstance && nextState.isInstance) {
             return false;
         }
-
         return true;
     }
 
-    handleSubmit(event) {
+    postData(body) {
+      if(this.state.isInstance) {
+        return datasets.updateInstanceEdition(this.props.params.instanceID, this.state.selectedEdition, body);
+      } else {
+        return datasets.updateVersion(this.props.params.datasetID, this.props.params.edition, this.props.params.version, body);
+      }
+    }
+
+    updateInstanceVersion(metaData) {
+      return this.postData(metaData).then(() => {
+        if(this.state.isInstance) {
+          datasets.getInstance(this.props.params.instanceID).then(response => {
+            this.props.dispatch(push(`${this.props.rootPath}/datasets/${this.props.params.datasetID}/editions/${response.edition}/versions/${response.version}/collection`));
```

Comment:
I don't think we need to do this redirect to the newly created version now because we go straight to the collection screen - happy to discuss this though because I might be missing something.

Reply:
Ignore me - this is absolutely fine!